topo/node/arista/arista.go

```diff
@@ -161,11 +161,11 @@ func (n *Node) CreateCRD(ctx context.Context) error {
 	for label, v := range proto.GetLabels() {
 		device.ObjectMeta.Labels[label] = v
 	}
-	for _, service := range proto.GetServices() {
+	for k, service := range proto.GetServices() {
 		if device.Spec.Services == nil {
 			device.Spec.Services = map[string]ceos.ServiceConfig{}
 		}
-		device.Spec.Services[service.Name] = ceos.ServiceConfig{
+		device.Spec.Services[node.DedupServiceNames(service, k)] = ceos.ServiceConfig{
```

Comment:
Can we address the service name before it is sent to the vendor code?  When the topology is imported we can check to see if service.Name is set, if not, check to see if there are any names and if so use the first one, and finally, if not, generate a name.  Then code like this does not have to change.  Or are you already working on this and I am just out of sequence?

Reply:
Yes I was working on it, fixed it to handle when topology is imported so that vendor code need not be changed.